_threat-model-common.adoc

```diff
@@ -16,31 +16,129 @@
-Similarly, when loading a configuration file over the network such as through HTTP, this should be configured to use TLS or a secure connection in general with strong authentication guarantees.
-This remote location must be kept safe from unauthorized modifications.
+Trusted Users::
++
+Application developers and administrators are considered **trusted** users.
+They have unrestricted access to all the features of the logging framework.
 
-For Java-based projects supporting JNDI or JMX,
-when configurations are modified through JMX, the JMX server should be safely configured to require authentication and a secure connection if being accessed over the network.
-When configurations are provided through JNDI, these should only use the `java` scheme for sharing configurations in a Java EE or Jakarta EE application service.
-JNDI-sourced configurations should not use other JNDI providers such as LDAP, DNS, or RMI, as all these providers are difficult to properly secure.
+Untrusted Users::
++
+All the other users are considered untrusted.
 
-[#threat-common-java-serialization]
-==== Java Object Serialization Stream Protocol
+[#threat-common-assets]
+== Data sources
 
-https://docs.oracle.com/javase/8/docs/platform/serialization/spec/protocol.html[Java Object Serialization Stream Protocol] should not be used to deserialize data from untrusted sources.
+Logging systems read data from multiple sources that are controlled by both trusted and untrusted users:
+
+Trusted Sources::
++
+* Log4cxx, Log4j and Log4net **trust** environment variables, configuration properties, and configuration files.
+To keep them secure:
+** It is up to the deployer to ensure that untrusted parties do not have write access to these resources.
+** It is up to the deployer to ensure that these resources are transmitted through a confidential channel.
+** Non-confidential channels such as HTTP or JMX are disabled by **default** to prevent their unintentional usage.
+** If configuration files use interpolation features such as
+https://logging.apache.org/log4j/2.x/manual/lookups.html[lookups], it is up to the deployer to ensure that only lookups from trusted sources are used.
+It is up to the programmer to document thread context keys that can be considered as trusted.
+
+* The logging frameworks **trust** that the objects passed to the log statements can be safely converted to strings:
+** These frameworks should not be used to log deserialized data from untrusted sources.
 See https://owasp.org/www-community/vulnerabilities/Deserialization_of_untrusted_data[the related OWASP guide] for details.
+
+* If parameterized logging is used, the format string is **trusted**:
+** Programmers **should** use compile time constants as format strings to prevent attackers from mangling messages.
```

Comment:
I'm not a native English speaker, but I'd replace `mangling` with `tampering`. (I might very well be wrong.)

Reply:
Fixed in https://github.com/apache/logging-site/pull/10/commits/6e96289d75abf0e6545e580d5c5004c6dffa6b35